File: blogeur_ia_python/blogueur_ia/embed_helpers.py
import re
import httpx

YOUTUBE_PATTERNS = [
    r"youtube\.com/watch\?v=([\w-]{11})",
    r"youtu\.be/([\w-]{11})",
    r"youtube\.com/embed/([\w-]{11})",
]
VIMEO_PATTERN = r"vimeo\.com/(\d+)"


def extraire_embed_video(url: str) -> dict | None:
    for pattern in YOUTUBE_PATTERNS:
        m = re.search(pattern, url)
        if m:
            return {"provider": "youtube", "embed_url": f"https://www.youtube.com/embed/{m.group(1)}"}
    m = re.search(VIMEO_PATTERN, url)
    if m:
        return {"provider": "vimeo", "embed_url": f"https://player.vimeo.com/video/{m.group(1)}"}
    return None


async def url_est_valide(url: str) -> bool:
    if not url:
        return False
    try:
        async with httpx.AsyncClient(timeout=8, follow_redirects=True) as http:
            resp = await http.head(url)
            if resp.status_code >= 400:
                resp = await http.get(url)
            return resp.status_code < 400
    except Exception:
        return False


def embed_maps(lat: float, lng: float, zoom: int = 15) -> str:
    return f"https://www.google.com/maps?q={lat},{lng}&z={zoom}&output=embed&hl=fr"

File: blogeur_ia_node/src/embedHelpers.ts
import axios from "axios";

const YOUTUBE_PATTERNS = [
  /youtube\.com\/watch\?v=([\w-]{11})/,
  /youtu\.be\/([\w-]{11})/,
  /youtube\.com\/embed\/([\w-]{11})/,
];
const VIMEO_PATTERN = /vimeo\.com\/(\d+)/;

export interface EmbedVideo {
  provider: "youtube" | "vimeo";
  embedUrl: string;
}

export function extraireEmbedVideo(url: string): EmbedVideo | null {
  for (const pattern of YOUTUBE_PATTERNS) {
    const m = url.match(pattern);
    if (m) return { provider: "youtube", embedUrl: `https://www.youtube.com/embed/${m[1]}` };
  }
  const m = url.match(VIMEO_PATTERN);
  if (m) return { provider: "vimeo", embedUrl: `https://player.vimeo.com/video/${m[1]}` };
  return null;
}

export async function urlEstValide(url: string): Promise<boolean> {
  if (!url) return false;
  try {
    let resp = await axios.head(url, { timeout: 8000, maxRedirects: 5, validateStatus: () => true });
    if (resp.status >= 400) {
      resp = await axios.get(url, { timeout: 8000, maxRedirects: 5, validateStatus: () => true });
    }
    return resp.status < 400;
  } catch {
    return false;
  }
}

export function embedMaps(lat: number, lng: number, zoom = 15): string {
  return `https://www.google.com/maps?q=${lat},${lng}&z=${zoom}&output=embed&hl=fr`;
}